pkg/common/jwtsvid/method.go

```diff
@@ -0,0 +1,87 @@
+package jwtsvid
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/asn1"
+	"math/big"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+var (
+	signingMethodES256 = &signingMethodECDSA{
+		SigningMethodECDSA: jwt.SigningMethodES256,
+	}
+)
+
+// signingMethodECDSA is a copy of the implementation of the JWT package
+// modified to accomodate both an *ecdsa.PrivateKey and a crypto.Signer based
+// key. It can be thrown away as soon as
+// https://github.com/dgrijalva/jwt-go/pull/236 is merged.
+type signingMethodECDSA struct {
+	*jwt.SigningMethodECDSA
+}
+
+func (m *signingMethodECDSA) Sign(signingString string, key interface{}) (string, error) {
+	// Get the signer
+	signer, ok := key.(crypto.Signer)
+	if !ok {
+		return "", jwt.ErrInvalidKeyType
+	}
+
+	// make sure the signer is for ECDSA
+	publicKey, ok := signer.Public().(*ecdsa.PublicKey)
+	if !ok {
+		return "", jwt.ErrInvalidKeyType
+	}
+
+	// Create the hasher
+	if !m.Hash.Available() {
+		return "", jwt.ErrHashUnavailable
+	}
+
+	hasher := m.Hash.New()
+	hasher.Write([]byte(signingString))
```

Comment:
are you referring to the return value of Write? in general, I agree, however in this case the standard hash functions don't ever return an error from Write. an error value is still part of the signature because 1) it is desirable for the hash interface to conform to the `io.Writer` interface, and 2) in general a hash implementation could fail (for example if it was backed by a kernel hash provider, e.g. AFALG). From the hash interface definition:

```go
type Hash interface {
        // Write (via the embedded io.Writer interface) adds more data to the running hash.
        // It never returns an error.
        io.Writer
```